tensorbay/opendataset/VGGFace2/loader.py

```diff
@@ -0,0 +1,224 @@
+                identity_meta.csv
+                loose_bb_test.csv
+                loose_bb_train.csv
+                loose_landmark_test.csv
+                loose_landmark_train.csv
+                attributes/
+                    01-Male.txt
+                    02-Black_Hair.txt
+                    ...
+
+            test/
+                n000001/
+                    0001_01.jpg
+                    0002_01.jpg
+                    ...
+                n000009/
+                ...
+
+            train/
+                n000002/
+                    0001_01.jpg
+                    ...
+                n000003/
+                ...
+
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+    dataset = Dataset(DATASET_NAME)
+
+    class_id_name, class_name_attrs = _get_classification_inform(root_path)
+    # class_id_name: {"n000001:"14th_Dalai_Lama",...}
+    # class_name_attrs: {"14th_Dalai_Lama":{"Sample_Num": 424, "Flag": 0, "Gender": 'm'],...}
+
+    dataset._catalog = Catalog.loads(  # pylint: disable=protected-access
+        _get_catalog(class_name_attrs)
+    )
+
+    box2d_file_cords = _get_box2d_cords(root_path)  # {"n000001/0001_01.jpg":[60,60,79,109],...}
+    box2d_file_attrs = _get_box2d_attrs(root_path)
+    # {'n000002/0039_01.jpg': {'01-Male': 0,'02-Black_Hair': 0,'03-Brown_Hair': 1,...},...}
+
+    ketpoints2d_file_keypoints = _get_keypoints2d_keypoints(root_path)
+    # {'n000001/0001_01.jpg': [[ 75.81253, 110.2077 ],[103.1778 , 104.6074 ],...],...}
+
+    for segment_name in _SEGMENTS:
+        segment = dataset.create_segment(segment_name)
+        with open(os.path.join(root_path, f"{segment_name}_list.txt"), "r", encoding="utf-8") as fp:
+            for line in fp.readlines():
+                info = line.rstrip("\n").split("/")
+                image_path = os.path.join(root_path, segment_name, info[0], info[1])
+                data = Data(image_path)
+                data.label.classification = Classification(
+                    category=class_id_name[info[0]],
+                    attributes=class_name_attrs[class_id_name[info[0]]],
+                )
+                data.label.box2d = [LabeledBox2D.from_xywh(*box2d_file_cords[line.rstrip("\n")])]
+                if box2d_file_attrs.get(line.rstrip("\n"), 0):
+                    data.label.box2d[0].attributes = box2d_file_attrs[line.rstrip("\n")]
+                data.label.keypoints2d = [
+                    LabeledKeypoints2D(ketpoints2d_file_keypoints[line.rstrip("\n")])
+                ]
+                segment.append(data)
+    return dataset
+
+
+def _get_catalog(class_name_attrs: Dict[str, Dict[str, object]]) -> Dict[str, Any]:
+    get_catalog = {}
+    class_names = class_name_attrs.keys()
+    nums = np.array([value["Sample_Num"] for value in class_name_attrs.values()])
```

Comment:
why we need `numpy` here?

Reply:
Because it is identified as a object, and the object type can not to be used to do some operations like minimize and maximize which will be used then.